lib/iex/lib/iex/introspection.ex

```diff
@@ -115,13 +126,37 @@ defmodule IEx.Introspection do
         end
         :ok
       else
-        :not_found
+        if has_callback?(mod, fun, arity) do
+          :behaviour_found
+        else
+          :not_found
+        end
       end
     else
       :no_docs
     end
   end
 
+  defp has_callback?(mod, fun) do
+    mod
+    |> Code.get_docs(:callback_docs)
+    |> find_callback_doc(fun)
+  end
+
+  defp has_callback?(mod, fun, arity) do
+    mod
+    |> Code.get_docs(:callback_docs)
+    |> find_callback_doc(fun, arity)
+  end
+
+  def find_callback_doc(docs, fun) do
+    Enum.any?(docs, &match?({{^fun, _}, _, _, _}, &1))
+  end
+
+  def find_callback_doc(docs, fun, arity) do
```

Comment:
`find_callback_doc/2,3` should be private, right? My personal opinion is that we could ditch these functions as we only use them once each and it would be clear enough but that's up to you and others in the team if they have a preference :).